src/arch/fdselecthand.cpp

```diff
@@ -63,8 +63,8 @@ void CFDSelectHandler::run() {
 
     if (retval > 0) {
       mSync.lock();
-      TConnectionContainer::Iterator itEnd(mConnectionsList.end());
-      for (TConnectionContainer::Iterator itRunner = mConnectionsList.begin(); itRunner != itEnd;) {
+      auto itEnd(mConnectionsList.end());
+      for (auto itRunner = mConnectionsList.begin(); itRunner != itEnd;) {
```

Comment:
Have you tested this loop with several different cases, especially with servers and clients where the other end is disconnecting, when several subscribers are listening? I don't think this loop will work in that case. That is the reason I have not yet replaced this single linked list with a std::vector. I'm not even sure if it is possible that we can use std::vector here.